docker/Dockerfile-build-backend.jvm

```diff
@@ -1,42 +1,33 @@
-####
-# This Dockerfile is used in order to build a container that runs the Quarkus application in JVM mode
-###
-
-# Build the project
 FROM registry.access.redhat.com/ubi8/openjdk-11:latest AS build
 USER root
-COPY . /home/jboss
+COPY . /home/jboss/
 WORKDIR /home/jboss
 RUN ./mvnw clean package -DskipTests --projects :checkstyle,:notifications-backend
 
-# Build the container
 FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
 
 ARG JAVA_PACKAGE=java-11-openjdk-headless
 ARG RUN_JAVA_VERSION=1.3.8
-
 ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en'
 
-# Install java and the run-java script
-# Also set up permissions for user `1001`
 RUN microdnf install openssl curl ca-certificates ${JAVA_PACKAGE} \
     && microdnf update \
     && microdnf clean all \
     && mkdir /deployments \
     && chown 1001 /deployments \
     && chmod "g+rwX" /deployments \
     && chown 1001:root /deployments \
-    && curl https://repo1.maven.org/maven2/io/fabric8/run-java-sh/${RUN_JAVA_VERSION}/run-java-sh-${RUN_JAVA_VERSION}-sh.sh -o /deployments/run-java.sh \
-    && chown 1001 /deployments/run-java.sh \
-    && chmod 540 /deployments/run-java.sh \
     && echo "securerandom.source=file:/dev/urandom" >> /etc/alternatives/jre/lib/security/java.security
 
 # Configure the JAVA_OPTIONS, you can add -XshowSettings:vm to also display the heap size.
 ENV JAVA_OPTIONS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
 
-COPY --from=build --chown=1001 /home/jboss/backend/target/quarkus-app/ /deployments/
+COPY --from=build --chown=1001 /home/jboss/backend/target/quarkus-app/lib/ /deployments/lib/
+COPY --from=build --chown=1001 /home/jboss/backend/target/quarkus-app/*.jar /deployments/
+COPY --from=build --chown=1001 /home/jboss/backend/target/quarkus-app/app/ /deployments/app/
+COPY --from=build --chown=1001 /home/jboss/backend/target/quarkus-app/quarkus/ /deployments/quarkus/
```

Comment:
Wasn't the old version enough?

Reply:
Both are doing the same, but splits it in four distinct layers so if there are application changes the library layers can be re-used. This is also the Quarkus way which everyone will get when generating a project via https://code.quarkus.io/

Reply:
I am +1 on splitting. I am not sure though if the CI pipeline really caches the old image or if we always start from scratch. For local builds it definitively helps. And I think when pushing to a repo, it is also quicker, as the lower layers will already be present and don't need to be re-pushed.